source/extensions/filters/common/lua/lua.h

```diff
@@ -69,6 +69,19 @@ namespace Lua {
   lua_pushnumber(state, val);                                                                      \
   lua_settable(state, -3);
 
+/**
+ * Get absl::string_view from Lua string. This checks if the argument at index is a string
+ * and build an absl::string_view from it.
+ * @param state the current Lua state.
+ * @param index the index of argument.
+ * @return absl::string_view of Lua string with proper string length.
+ **/
+inline absl::string_view getStringViewFromLuaString(lua_State* state, int arg) {
+  size_t input_size;
+  const char* input = luaL_checklstring(state, arg, &input_size);
```

Comment:
Can you clarify that if this fails, we return an empty string?

Reply:
@dio From the source code of LuaJIT, `luaL_checklstring` will throws an error if this is invalid.

Reply:
`luaL_checklstring` is similar with `luaL_checkstring`. It also return the size by an additional parameter.

Reply:
Yes. I know that it has been the default behavior for these `luaL_check*` APIs. I think we can have a comment on this.

Reply:
> a sample output in the log

Is it means catch the error, and print error msg in the log?

Reply:
A comment is enough for now. For example:

```
  // When the argument at index in Lua state is not a string, for example, giving a table to
  // logTrace (which uses this function under the hood), Lua script exits with an error like the
  // following: "[string \"...\"]:3: bad argument #1 to 'logTrace' (string expected, got table)".
  // However, "luaL_checklstring" accepts a number as its argument and implicitly converts it to a
  // string, since Lua provides automatic conversion between string and number values at run time
  // (https://www.lua.org/manual/5.1/manual.html#2.2.1).

```

Optionally you can add a test, for example in lua_filter_test.cc as it is easier (?). This also depicts the current behavior of this filter, which can be useful when one wants to implement a feature like: `fail-close/fail-open` https://github.com/envoyproxy/envoy/issues/14980.

```c++
// Script logging a table instead of the expected string.
TEST_F(LuaHttpFilterTest, LogTableInsteadOfString) {
  const std::string LOG_TABLE{R"EOF(
    function envoy_on_request(request_handle)
      request_handle:logTrace({})
    end
  )EOF"};

  InSequence s;
  setup(LOG_TABLE);

  Http::TestRequestHeaderMapImpl request_headers{{":path", "/"}};
  EXPECT_CALL(*filter_, scriptLog(spdlog::level::err, StrEq("[string \"...\"]:3: bad argument #1 to 'logTrace' (string expected, got table)")));
  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, true));
}

```